### PR TITLE
Add path GC for BE

### PR DESCRIPTION
**Pull request by kangpinghuang**
1. Add path gc
2. Optimize rowset lazy load
3. Fix some NPE bugs

### CODE REVIEW

File: be/src/olap/storage_engine.cpp [hunk 1]
```diff
@@ -1021,4 +987,232 @@ OLAPStatus StorageEngine::execute_task(EngineTask* task) {
     }
 }
 
+void StorageEngine::add_check_paths(std::set<std::string> paths) {
```
Comment (chaoyli) [inline be/src/olap/storage_engine.cpp]: std::set<std::string>&

File: be/src/olap/tablet_manager.h [hunk 2]
```diff
@@ -108,7 +108,9 @@ class TabletManager {
     TabletSharedPtr find_best_tablet_to_compaction(CompactionType compaction_type);
 
     // Get tablet pointer
-    TabletSharedPtr get_tablet(TTabletId tablet_id, SchemaHash schema_hash, bool load_tablet = true);
+    TabletSharedPtr get_tablet(TTabletId tablet_id, SchemaHash schema_hash);
+
+    bool get_tablet_id_and_schema_hash_from_path(std::string path, TTabletId* tablet_id, TSchemaHash* schema_hash);
```
Comment (chaoyli) [inline be/src/olap/tablet_manager.h]: std::string&

File: be/src/olap/tablet.h [hunk 3]
```diff
@@ -63,6 +63,8 @@ class Tablet : public std::enable_shared_from_this<Tablet> {
     Tablet(TabletMeta* tablet_meta, DataDir* data_dir);
     ~Tablet();
 
+    static bool is_valid_tablet_path(std::string path);
+
```
Comment (chaoyli) [inline be/src/olap/tablet.h]: std::string&

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/storage_engine.cpp [hunk 4]
```diff
@@ -1021,4 +987,232 @@ OLAPStatus StorageEngine::execute_task(EngineTask* task) {
...
+    uint32_t interval = config::path_gc_check_interval_second;
+    if (interval <= 0) {
+        LOG(WARNING) << "path gc thread check interval config is illegal:" << interval
+            << "will be forced set to half hour";
+        interval = 1800; // 0.5 hour
+    }
+
+    while (true) {
+        LOG(INFO) << "try to perform path gc!";
+        CgroupsMgr::apply_system_cgroup();
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/storage_engine.cpp]: OK

File: be/src/olap/storage_engine.cpp (hunk 1 above)
Comment (kangpinghuang) [reply be/src/olap/storage_engine.cpp]: ok

File: be/src/olap/tablet_manager.h (hunk 2 above)
Comment (kangpinghuang) [reply be/src/olap/tablet_manager.h]: ok

**Pull request merged**